File: services/admin-api/internal/domain/common/audit.go
package common

import "time"

// AuditLog 是审计日志值对象（写后只读）。
type AuditLog struct {
	ID           int64
	ActorID      int64
	Action       string
	ResourceType string
	ResourceID   string
	Env          Environment
	Detail       AuditDetail
	CreatedAt    time.Time
}

// AuditDetail 记录关键 before/after 与请求摘要。
type AuditDetail struct {
	Summary string            `json:"summary,omitempty"`
	Before  map[string]any    `json:"before,omitempty"`
	After   map[string]any    `json:"after,omitempty"`
	Changed []string          `json:"changed,omitempty"`
	Extra   map[string]any    `json:"extra,omitempty"`
	Request *AuditRequestMeta `json:"request,omitempty"`
}

// AuditRequestMeta 保存请求元信息，用于审计追踪。
type AuditRequestMeta struct {
	IP        string `json:"ip,omitempty"`
	UserAgent string `json:"userAgent,omitempty"`
	RequestID string `json:"requestId,omitempty"`
	Method    string `json:"method,omitempty"`
	Path      string `json:"path,omitempty"`
}

File: apps/admin-web/src/api/modules/audit.ts
import { request } from "@/api/http";

export type AuditEnvironment = "develop" | "sandbox" | "production" | string;

export interface Paginated<T> {
  items: T[];
  page: number;
  pageSize: number;
  total: number;
}

export interface AuditOperator {
  id: string;
  userName: string;
  displayName: string;
}

export interface AuditRequestMeta {
  ip?: string;
  userAgent?: string;
  requestId?: string;
  method?: string;
  path?: string;
}

export interface AuditDetailPayload {
  summary?: string;
  before?: Record<string, unknown>;
  after?: Record<string, unknown>;
  changed?: string[];
  extra?: Record<string, unknown>;
  request?: AuditRequestMeta;
  [key: string]: unknown;
}

export interface AuditLogItem {
  id: string;
  actorId: string;
  operator: AuditOperator | null;
  action: string;
  resourceType: string;
  resourceId: string;
  env: AuditEnvironment;
  detail: AuditDetailPayload;
  createdAt: string;
}

export interface ListAuditLogsQuery {
  env?: AuditEnvironment;
  action?: string;
  resourceType?: string;
  resourceId?: string;
  operator?: string;
  operatorKeyword?: string;
  from?: string;
  to?: string;
  keyword?: string;
  page?: number;
  pageSize?: number;
  sort?: "createdAt" | "-createdAt";
}

export interface AuditLogFacets {
  envs: string[];
  actions: string[];
  resourceTypes: string[];
}

function buildQuery(params: Record<string, unknown>): string {
  const search = new URLSearchParams();
  for (const [key, value] of Object.entries(params)) {
    if (value !== undefined && value !== null && value !== "") {
      search.append(key, String(value));
    }
  }
  const query = search.toString();
  return query ? `?${query}` : "";
}

export function listAuditLogs(query: ListAuditLogsQuery = {}): Promise<Paginated<AuditLogItem>> {
  return request<Paginated<AuditLogItem>>(`/api/admin/audit-logs${buildQuery({ ...query })}`);
}

export function getAuditLogDetail(id: string): Promise<AuditLogItem> {
  return request<AuditLogItem>(`/api/admin/audit-logs/${encodeURIComponent(id)}`);
}

export function listAuditLogFacets(): Promise<AuditLogFacets> {
  return request<AuditLogFacets>("/api/admin/audit-logs/facets");
}
